qa/workunits/rgw/test_rgw_datacache.py

```diff
@@ -170,33 +170,46 @@ def main():
     out = exec_cmd('radosgw-admin object stat --bucket=%s --object=%s' % (BUCKET_NAME, FILE_NAME))
 
     json_op = json.loads(out)
-    cached_object_name = json_op['manifest']['prefix']
-    log.debug("Cached object name is: %s", cached_object_name)
+    bucket_marker = json_op['manifest']['tail_placement']['bucket']['marker']
+    object_name = json_op['name']
+
+    # get num parts of the cached object, 7MB is the size of the object, 4MB is the chunk size used in d3n filter
+    object_size = 7340032
+    chunk_size = 4194304
+    num_parts = int(object_size/chunk_size)
+    if object_size%chunk_size > 0:
+        num_parts = num_parts + 1
+    log.info("Num parts is: %s", num_parts)
 
     # check that the cache is enabled (does the cache directory empty)
     out = exec_cmd('find %s -type f | wc -l' % (cache_dir))
     chk_cache_dir = int(get_cmd_output(out))
     log.debug("Check cache dir content: %s", chk_cache_dir)
-    if chk_cache_dir == 0:
+    if chk_cache_dir != num_parts:
         log.info("NOTICE: datacache test object not found, inspect if datacache was bypassed or disabled during this check.")
```

Comment:
think that the `if chk_cache_dir == 0:` is a relevant test case in itself (and the log message points to a specific condition that is common and important to distinct)
may suggest possibly to add another test immediately afterwards, something along the lines of for example
```
if chk_cache_dir != num_parts:
        log.info("ERROR: not all the parts of the datacache test object were found in the cache.")
```

Reply:
ok, makes sense